docs/essentials/toolbars-and-globals.md

```diff
@@ -0,0 +1,272 @@
+---
+title: 'Toolbars & globals'
+---
+
+<div style="background-color:#F8FAFC">
+TODO: cleanup commented out text once it's vetted.
+</div>
+
+By default, Storybook ships with toolbar items to control the [viewport](./viewport) and [background](./backgrounds) the story renders in. You can also create your own toolbar items which control special “globals” which you can then read to create [decorators](../writing-stories/decorators) to control story rendering.
+
+## Globals
+
+Globals in Storybook represent “global” (as in not story-specific) inputs to the rendering of the story. As they aren’t specific to the story, they aren’t passed in the `args` argument to the story function (although they are accessible as `context.globals`), but typically you use them in decorators which apply to all stories.
+
+When the globals change, the story re-renders and the decorators rerun with the new values. The easiest way to change globals is to create a toolbar item for them. 
+
+Let’s see how.
+
+## Global Types and the toolbar annotation
+
+Storybook has a simple, declarative syntax for configuring toolbar menus. In your [`.storybook/preview.js`](../configure/overview#configure-story-rendering), you can add your own toolbars by creating `globalTypes` with a `toolbar` annotation: 
+
+<!--  You can add your own toolbars by adding `globalTypes` with a `toolbar` annotation, in [`.storybook/preview.js`](../configure/overview#configure-story-rendering): -->
+
+```js
+// .storybook/preview.js
+
+export const globalTypes = {
+  theme: {
+    name: 'Theme'
+    description: 'Global theme for components',
+    defaultValue: 'light',
+    toolbar: {
+      icon: 'circlehollow',
+      // array of plain string values or MenuItem shape (see below)
+      items: ['light', 'dark'],
+    },
+  },
+};
+```
+
+> As globals are *global* you can *only* set `globalTypes` in [`.storybook/preview.js`](../configure/overview#configure-story-rendering).
+
+When you start your Storybook, you should see a new dropdown in your toolbar with options `light` and `dark`.
+
+### Create a decorator
+
+We have a `global` defined, let's wire it up! We can consume our new `theme` global in a decorator using the `context.globals.theme` value.
+
+<!-- Now, let's wire it up! We can consume our new `theme` global in a decorator using the `context.globals.theme` value. -->
+
+For example, suppose you are using `styled-components`. You can add a theme provider decorator to your [`.storybook/preview.js`](../configure/overview#configure-story-rendering) config:
+
+<div style="background-color:#F8FAFC">
+TODO: ask tom/michael if this needs to be typescript or can be js like below. as we've been showing this in js and ts is added without warning
```

Comment:
Hmm, not sure why it was TS, JS is better 👍

Reply:
js it is then. If required we have the ts option and we can add it a later time.